src/transformers/rc4.ts

```diff
@@ -0,0 +1,101 @@
+/// <reference path='../../../third_party/sha1/sha1.d.ts' />
+/// <reference path='../../../third_party/simple-rc4/simple-rc4.d.ts' />
+/// <reference path='../../../third_party/typings/browser.d.ts' />
+
+import logging = require('../logging/logging');
+import rc4 = require('simple-rc4');
+import sha1 = require('crypto/sha1');
+import transformer = require('./transformer');
+
+const log = new logging.Log('rc4 transformer');
+
+// Accepted in serialised form by configure().
+export interface Config {
+  key: string
+}
+
+const KEY_LENGTH_BYTES = 16;
+const R_LENGTH_BYTES = 8;
+const TRUNCATED_IV_LENGTH_BYTES = 8;
+
+// Creates a sample (non-random) config, suitable for testing.
+export function sampleConfig(): Config {
+  return {
+    key: new Buffer(KEY_LENGTH_BYTES).fill(0).toString('hex')
+  };
+}
+
+// Fast, uniformly random transformer with minimal length expansion, via RC4.
+// From a proposal by kpdyer.
+//
+// Terminology:
+//  - K: 128-bit session key, negotiated through the signalling channel
+//  - RANDOM(N): a cryptographically secure N-byte string
+//  - X[n,...,m]: the (m - n + 1) bytes starting at byte-index n of X
+//  - |X|: length, in bytes, of X
+//
+// Pseudo-code:
+//   def Encrypt(K, P):
+//     R = RANDOM(8)
+//     IV = SHA1(K || R)
+//     RET R || RC4(IV[0,...,7], P)
+//
+//   def Decrypt(K1, C):
+//     R = C[0,...,7]
+//     IV = SHA1(K1 || R)
+//     RET RC4(IV[0,...,7], C[8,...,|C | -1])
+export class Rc4Transformer implements transformer.Transformer {
+  private key_: Buffer;
+
+  public constructor() {
+    this.configure(JSON.stringify(sampleConfig()));
+  }
+
+  public configure = (json: string): void => {
+    try {
+      const config = <Config>JSON.parse(json);
+      if (config.key === undefined) {
+        throw new Error("must set key parameter");
+      }
+      const key = new Buffer(config.key, 'hex');
+      if (key.byteLength !== KEY_LENGTH_BYTES) {
```

Comment:
Are there any unit tests associated with this code? Specifically, I'm curious how Buffer knows the length of config.key is KEY_LENGTH_BYTES. In some languages a call like this might trim leading zeros. Can you add some tests to verify potential failure cases? e.g., the all-0 key, keys with leading 0s, etc.